src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -268,3 +311,27 @@ function validate_connectivity(M, nodes, bus_lookup)
     end
     return connected
 end
+
+# this function extends the PowerModels.jl implementation to accept a System
+function calc_connected_components(sys::System)
```

Comment:
all of these functions will have to accept kwargs... to define the connectivity check method